server/src/main/java/com/thoughtworks/go/server/messaging/plugin/AgentStatusChangeNotifier.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.messaging.plugin;
+
+import com.thoughtworks.go.domain.AgentInstance;
+import com.thoughtworks.go.domain.notificationdata.AgentNotificationData;
+import com.thoughtworks.go.listener.AgentStatusChangeListener;
+import com.thoughtworks.go.plugin.access.notification.NotificationExtension;
+import com.thoughtworks.go.plugin.access.notification.NotificationPluginRegistry;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+public class AgentStatusChangeNotifier implements AgentStatusChangeListener{
+    private NotificationPluginRegistry notificationPluginRegistry;
+    private PluginNotificationQueue pluginNotificationQueue;
+
+    @Autowired
+    public AgentStatusChangeNotifier(NotificationPluginRegistry notificationPluginRegistry, PluginNotificationQueue pluginNotificationQueue) {
+        this.notificationPluginRegistry = notificationPluginRegistry;
+        this.pluginNotificationQueue = pluginNotificationQueue;
+    }
+
+    @Override
+    public void onAgentStatusChange(AgentInstance agentInstance) {
+        if (isAnyPluginInterestedInAgentStatus()) {
+            AgentNotificationData agentNotificationData = notificationDataFrom(agentInstance);
+            pluginNotificationQueue.post(new PluginNotificationMessage(NotificationExtension.AGENT_STATUS_CHANGE_NOTIFICATION, agentNotificationData));
+        }
+    }
+
+    private AgentNotificationData notificationDataFrom(AgentInstance agentInstance) {
+        return new AgentNotificationData(agentInstance.getUuid(),
+                agentInstance.getHostname(),
+                elasticAgentId(agentInstance),
```

Comment:
What do we plan to do with the `elasticAgentId`?

Reply:
Reason for asking this is `elasticAgentId` by itself is not very useful. If all we care about is if the agent is elastic, then may be we could pass along a flag for that. If we do care about the `elasticAgentId` then most likely we would also care about the `elasticPluginId` too.

Reply:
I don't see a use case for using the elastic_agent_id apart from being able to distinguish between elastic and non-elastic agents. You are right that just the elastic_agent_id might not be of much use.
Changed this to pass a flag `is_elastic` for now.